wad2project/rotten_potatoes/models.py

```diff
@@ -3,25 +3,27 @@
 from django.contrib.auth.models import User
 from datetime import datetime
 
+
 # Create your models here.
 class UserProfile(models.Model):
     user = models.OneToOneField(User, on_delete=models.CASCADE)
     profile_pic = models.ImageField(upload_to="profile_images", blank=True)
     description = models.TextField(max_length=1024, default="")
 
-    #boolean flag for identifying producers
+    # boolean flag for identifying producers
     producer = models.BooleanField(default=False)
 
     def __str__(self):
         return self.user.username
 
+
 class Movie(models.Model):
     name = models.CharField(max_length=128)
     release_date = models.DateField(null=True)
     actors = models.CharField(max_length=256)
     genre = models.CharField(max_length=256)
     # Producer ForeignKey
-    producer = models.ForeignKey(UserProfile, on_delete=models.CASCADE)
+    producer = models.ForeignKey(UserProfile, on_delete=models.CASCADE, null=True, blank=True)
```

Comment:
Should it be possible for a movie to have no producer?

Reply:
There was a bug and this solved it. I believe the problem was that when saving data from form, I must not commit first time I save it (check the movie view) because I have to manually set some fields. One of those fields is the producer. Since its primary key when I save it for the first time it throws Null Exception since no data were given. Thus, allowing it to be null solved the problem.